blob/azureblob/azureblob.go

```diff
@@ -419,11 +420,14 @@ func NewPipeline(credential azblob.Credential, opts azblob.PipelineOptions) pipe
 // write and delete operations on objects within it.
 // See https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction.
 type bucket struct {
-	name         string
-	pageMarkers  map[string]azblob.Marker
-	serviceURL   *azblob.ServiceURL
-	containerURL azblob.ContainerURL
-	opts         *Options
+	name                  string
+	pageMarkers           map[string]azblob.Marker
+	serviceURL            *azblob.ServiceURL
+	containerURL          azblob.ContainerURL
+	opts                  *Options
+	credentialExpiration  *time.Time
```

Comment:
I think it's more standard to use `time.Time` and check for `ZeroTime` instead of `nil`. In this case, since "now" will be after the zero time, you don't even need that special check (on line 845).